### PR TITLE
[MM-44873] Screen reader not announcing the current state of user

### PR DESCRIPTION
**Pull request by babinderrathi**
#### Summary
screen reader not announcing the current state of user 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44873

#### Related Pull Requests
none

#### Screenshots
|  before  |

https://user-images.githubusercontent.com/102028591/190590603-057f6610-91bb-459f-b477-96913fabf50b.mp4

|  after  |

https://user-images.githubusercontent.com/102028591/191520032-978b076e-c4bf-4179-9334-9f56db2b869e.mp4

#### Release Note
```release-note
when we hover over the current status icon, the Screen Reader does not announce the status of the user. Now it announcing the current status and also tell user to open and change the status.
```

### CODE REVIEW

**Comment by cwarnermm** [comment]
Does the phrase "You can change status by opening button" follow existing phrasing elsewhere across the UI? I propose updating the audible text to specify the specific UI element the user needs to select to change their user status. Something along the lines of "You can change status by selecting your profile picture".

**Comment by hmhealey** [comment]
^ @laneycs

I'm not sure exactly how we'd want this to work since we probably want something a bit more concise here. Something like "current status is away, set status button". Also, I think the "button" part at the end can't be changed since the screen reader adds that on itself

File: components/status_dropdown/status_dropdown.tsx [hunk 1]
```diff
@@ -398,6 +400,9 @@ export default class StatusDropdown extends React.PureComponent<Props, State> {
 
         const customStatusComponent = this.renderCustomStatus(isStatusSet);
 
+        const menuAriaLabelValue = this.props.status;
+        const menuAriaLabeltext = intl.formatMessage({id: 'status_dropdown.menuAriaLabel.text', defaultMessage: 'user status is set to ' + menuAriaLabelValue + ' you can change status by opening'});
```
Comment (hmhealey) [inline components/status_dropdown/status_dropdown.tsx:404]: The exact text still needs to be finalized, but to add a value like the status into a translated piece of text like this, you need to use the `values` option to do that so it'd look like
```suggestion
        const menuAriaLabeltext = intl.formatMessage({
            id: 'status_dropdown.menuAriaLabel.text',
            defaultMessage: 'user status is set to {status} you can change status by opening',
            values: {status: this.props.status},
        });
```

**Review by hmhealey — CHANGES_REQUESTED** [review]
After making this change (and any other text changes), you'll have to run `make i18n-extract` to update `en.json` with the default text that's being used here

**Comment by laneycs** [comment]
@hmhealey @cwarnermm Just tested this with Voiceover on macOS Monterey. The existing copy reads and speaks as follows:

1. When the user profile is on:hover, "User profile image, image, banner."
2. When the user profile is selected to view the dropdown menu: "You are currently on an image. To begin interacting with the contents of this image, press Control-Optin-Shift-Down Arrow."

I believe this user flow should be more descriptive and follow similar existing copy in the navigation menu as follows:

1. When the user profile is selected, "You are currently on the user profile image. Your status is currently set to ______. Select to update your profile settings and status."
2. When the user profile is selected: "You are currently on the user profile menu. To begin interacting with this menu, press Control-Option-Shift-Down Arrow."
3. As the user navigates through the profile menu:

- "You are currently on user profile settings. Select to open the user profile settings modal."
- "You are currently on user status: Custom. Select to set a custom status and emoji."
- "You are currently on user status: Online. Select to set your status to Online."
- "You are currently on user status: Away. Select to set your status to Away."
- "You are currently on user status: Do not Disturb. Select to set your status to Do not Disturb and choose a timeframe for disabling your notifications."
- "You are currently on user status: Offline. Select to set your status to Offline."

File: components/status_dropdown/status_dropdown.test.tsx [hunk 2]
```diff
@@ -169,7 +169,7 @@ describe('components/StatusDropdown', () => {
             <StatusDropdown {...props}/>,
         );
 
-        expect(wrapper.find('.status-dropdown-menu__clear-container').exists()).toBe(true);
+        expect(wrapper.find('.status-dropdown-menu__clear-container').exists()).toBe(false);
```
Comment (nevyangelova) [inline components/status_dropdown/status_dropdown.test.tsx:172]: Does this snapshot need to be updated? As far as i can see the UI shouldn't change here

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: components/status_dropdown/status_dropdown.test.tsx (hunk 2 above)
Comment (babinderrathi) [reply components/status_dropdown/status_dropdown.test.tsx:172]: Okay,  I got it. I will look into it.

**Review by nevyangelova — APPROVED** [review]
LGTM!

**Review by laneycs — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks for that change. Apologies for the delay in re-reviewing this PR

**Comment by jgilliam17** [comment]
Thanks @babinderrathi 
Significant number of cypress tests failed. See full report [here](https://mattermost-cypress-report.s3.amazonaws.com/247973-5ffbea6-onprem-ent-webapp-pr-11145-mm-ee-test:5ffbea6/mochawesome.html) 
Can you please take a look and fix. Thanks 🙂

**Comment by jgilliam17** [comment]
@babinderrathi  When testing manually, JAWS reader is announcing user status - that ~looks~ sounds good.
I re-ran the [e2es](https://automation-dashboard.vercel.app/cycles/ce383ba6-edc1-41e5-b8a9-1ab46676ae16) after updating the branch and still seeing high number of failures. Let me know if you need help identifying failed tests.
Additionally if you have any questions or need help with Cypress please ask in our [QA channel](https://community.mattermost.com/core/channels/quality-assurance) on community. Thanks 🙂

**Comment by hmhealey** [comment]
Hi, @babinderrathi. If you have any questions or if there's anything we could help out with, feel free to let us know

**Review by jgilliam17 — APPROVED** [review]
Thanks @babinderrathi for fixing the e2e tests 
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/257199-d5388ef-onprem-ent-webapp-pr-11145-mm-ee-test:d5388ef/mochawesome.html) - no PR related failures. Looks good to merge.

**Comment by jgilliam17** [comment]
@hmhealey Can you please approve the last workflow and merge? Thanks 🙂

**Comment by amyblais** [comment]
@hmhealey @nevyangelova Can you help manually cherry-pick this to ``release-7.1`` branch?

**Pull request merged**